core/src/main/scala/org/apache/spark/ui/SparkUI.scala

```diff
@@ -141,6 +141,7 @@ private[spark] object SparkUI {
   val DEFAULT_POOL_NAME = "default"
   val DEFAULT_RETAINED_STAGES = 1000
   val DEFAULT_RETAINED_JOBS = 1000
+  val DEFAULT_RETAINED_TASKS = 10000
```

Comment:
As for the why, the issue is when the history server is full of applications with large task amounts, there is no issue with just one application having a significant number of tasks by itself.